### PR TITLE
Move repeat views to utility

### PR DESCRIPTION
**Pull request by MitraDarja**
Resolves https://github.com/seqan/product_backlog/issues/354
Resolves https://github.com/seqan/product_backlog/issues/353

### CODE REVIEW

File: test/unit/utility/views/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,5 @@
 add_subdirectories()
 
+seqan3_test(repeat_n_test.cpp)
+seqan3_test(repeat_test.cpp)
 seqan3_test (zip_test.cpp)
```
Comment (Irallia) [inline test/unit/utility/views/CMakeLists.txt:5]: As you touch this file, you could remove this space
```suggestion
seqan3_test(zip_test.cpp)
```

**Review by Irallia — APPROVED** [review]
LGFM, just a space you could remove. 💅

File: include/seqan3/alignment/matrix/detail/score_matrix_single_column.hpp [hunk 2]
```diff
@@ -18,7 +18,7 @@
 #include <seqan3/alignment/matrix/detail/affine_cell_proxy.hpp>
 #include <seqan3/alignment/matrix/detail/matrix_coordinate.hpp>
 #include <seqan3/range/container/aligned_allocator.hpp>
-#include <seqan3/range/views/repeat_n.hpp>
+#include <seqan3/utility/views/repeat_n.hpp>
 #include <seqan3/utility/detail/exposition_only_concept.hpp>
 #include <seqan3/utility/simd/concept.hpp>
 #include <seqan3/utility/views/zip.hpp>
```
Comment (marehr) [inline include/seqan3/alignment/matrix/detail/score_matrix_single_column.hpp:24]:
```suggestion
#include <seqan3/utility/detail/exposition_only_concept.hpp>
#include <seqan3/utility/simd/concept.hpp>
#include <seqan3/utility/views/repeat_n.hpp>
#include <seqan3/utility/views/zip.hpp>
```

File: include/seqan3/io/detail/record.hpp [hunk 3]
```diff
@@ -17,8 +17,8 @@
 #include <seqan3/std/ranges>
 
 #include <seqan3/io/record.hpp>
-#include <seqan3/range/views/repeat.hpp>
 #include <seqan3/utility/tuple/concept.hpp>
+#include <seqan3/utility/views/repeat.hpp>
 #include <seqan3/utility/type_list/traits.hpp>
```
Comment (marehr) [inline include/seqan3/io/detail/record.hpp:22]:
```suggestion
#include <seqan3/utility/type_list/traits.hpp>
#include <seqan3/utility/views/repeat.hpp>
```

File: include/seqan3/utility/views/repeat.hpp [hunk 4]
```diff
@@ -0,0 +1,368 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ * \brief  Provides the views::repeat_view.
```
Comment (marehr) [inline include/seqan3/utility/views/repeat.hpp:10]:
```suggestion
 * \brief  Provides the seqan3::views::repeat.
```

File: include/seqan3/utility/views/repeat.hpp [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+ *
+ * // std::string str1 = seqan3::views::repeat_n('A', 5) | seqan3::views::to<std::string>;
+ * // same as
+ * // std::string str2{5, 'A'};
+ *
+ * \hideinitializer
+ *
+ * \experimentalapi{Experimental since version 3.1.}
+ *
+ * TODO move to seqan3/utility/views
```
Comment (marehr) [inline include/seqan3/utility/views/repeat.hpp:363]:
```suggestion
```

File: include/seqan3/utility/views/repeat_n.hpp [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+ *
+ * ### Example
+ *
+ * \include test/snippet/utility/views/repeat_n.cpp
+ *
+ * \hideinitializer
+ *
+ * \experimentalapi{Experimental since version 3.1.}
+ *
+ * TODO move to seqan3/utility/views
```
Comment (marehr) [inline include/seqan3/utility/views/repeat_n.hpp:97]:
```suggestion
```

File: test/unit/utility/views/CMakeLists.txt [hunk 7]
```diff
@@ -1,3 +1,5 @@
 add_subdirectories()
 
+seqan3_test(repeat_n_test.cpp)
+seqan3_test(repeat_test.cpp)
```
Comment (marehr) [inline test/unit/utility/views/CMakeLists.txt:4]:
```suggestion
seqan3_test (repeat_n_test.cpp)
seqan3_test (repeat_test.cpp)
```

File: test/unit/utility/views/CMakeLists.txt (hunk 1 above)
Comment (marehr) [reply test/unit/utility/views/CMakeLists.txt:5]: In cmake we do space before `(` xD

**Review by marehr — COMMENTED** [review]
thank you; just some small things

File: include/seqan3/utility/views/repeat.hpp [hunk 8]
```diff
@@ -0,0 +1,365 @@
...
+ * ### Example
+ *
+ * \include test/snippet/utility/views/repeat.cpp
+ *
+ * \todo add snippet for code smells like this:
+ *
+ * // std::string str1 = seqan3::views::repeat_n('A', 5) | seqan3::views::to<std::string>;
+ * // same as
+ * // std::string str2{5, 'A'};
+ *
```
Comment (marehr) [inline include/seqan3/utility/views/repeat.hpp:358]:
```suggestion
```

File: include/seqan3/utility/views/repeat.hpp [hunk 9]
```diff
@@ -0,0 +1,359 @@
...
+ * See the \link views views submodule documentation \endlink for detailed descriptions of the view properties.
+ *
+ * \attention The given value to repeat is always **copied** into the range.
+ *
+ * ### Example
+ *
+ * \include test/snippet/utility/views/repeat.cpp
+ *
+ * \hideinitializer
+ *
```
Comment (marehr) [inline include/seqan3/utility/views/repeat.hpp:354]:
```suggestion
```

**Review by marehr — APPROVED** [review]

**Pull request merged**